### PR TITLE
Make pxf cluster sync delete files

### PR DESCRIPTION
**Pull request by oliverralbertini**
Previously pxf cluster sync was not deleting files from PXF_CONF that
had been deleted on master. By adding the `--delete` flag to the rsync
commands, we delete files from remote hosts.

Authored-by: Oliver Albertini <oalbertini@pivotal.io>

### CODE REVIEW

**Comment by oliverralbertini** [comment]
Confirmed that this works on `phoenix` cluster.

**Comment by oliverralbertini** [comment]
@denalex The flag you recommended is working, to avoid deletion, you can use `--non-destructive`. I am open to suggestions on the name.

**Comment by oliverralbertini** [comment]
```
pxf cluster sync --help
Sync PXF configs from master to standby master and all segment hosts

Usage: pxf cluster <command>
       pxf cluster {-h | --help}

Flags:
  -h, --help              help for sync
  -n, --non-destructive   Don't delete files on remote host
```

File: cli/go/src/pxf-cli/cmd/cluster.go [hunk 1]
```diff
@@ -54,6 +56,7 @@ func init() {
 	clusterCmd.AddCommand(startCmd)
 	clusterCmd.AddCommand(stopCmd)
 	clusterCmd.AddCommand(statusCmd)
+	syncCmd.Flags().BoolVarP(&NonDestructiveSync, "non-destructive", "n", false, "Don't delete files on remote host")
```
Comment (frankgh) [inline cli/go/src/pxf-cli/cmd/cluster.go]: maybe be more explicit here:
```suggestion
	syncCmd.Flags().BoolVarP(&NonDestructiveSync, "non-destructive", "n", false, "Do not delete files on segment hosts that have been deleted on master host")
```

File: cli/go/src/pxf-cli/cmd/pxf.go [hunk 2]
```diff
@@ -0,0 +1,205 @@
+package cmd
```
Comment (frankgh) [inline cli/go/src/pxf-cli/cmd/pxf.go]: I assume this is the file that was moved from `.../pxf/` to `cmd` correct?

Comment (oliverralbertini) [reply cli/go/src/pxf-cli/cmd/pxf.go]: yes

**Comment by oliverralbertini** [comment]
I've incorporated @lisakowen 's feedback; `pxf cluster sync` only deletes files if user passes `-d` or `--delete`.

```
$ pxf cluster sync --help
Sync PXF configs from master to standby master and all segment hosts. Use --delete to delete extraneous remote files

Usage: pxf cluster <command>
       pxf cluster {-h | --help}

Flags:
  -d, --delete   Delete extraneous files on remote host
  -h, --help     help for sync

$ pxf cluster --help
Perform <command> on each segment host in the cluster

Usage: pxf cluster <command>
       pxf cluster {-h | --help}

List of Commands:
  init        Initialize the PXF server instances on master, standby master, and all segment hosts
  reset       Reset PXF (undo initialization) on all segment hosts
  restart     Restart the PXF server on all segment hosts
  start       Start the PXF server instances on all segment hosts
  status      Get status of PXF servers on all segment hosts
  stop        Stop the PXF server instances on all segment hosts
  sync        Sync PXF configs from master to standby master and all segment hosts. Use --delete to delete extraneous remote files

Flags:
  -h, --help   help for cluster

Use "pxf cluster [command] --help" for more information about a command.
```

**Comment by lisakowen** [comment]
thanks, @oliverralbertini .  question:  we support `pxf sync <gphost>` from the master.  just want to verify that the new `{-d | --delete}` option is valid in the single-host-sync case.

**Comment by oliverralbertini** [comment]
> thanks, @oliverralbertini . question: we support `pxf sync <gphost>` from the master. just want to verify that the new `{-d | --delete}` option is valid in the single-host-sync case.

that's a great point. For `pxf init` we have the user pass the flag `-y` after the command. We can do the same for `pxf sync`, so with a8f3eb6eb70c3244627a1a060b2915038980cd91 it's  working like this `pxf sync -d host` or `pxf sync --delete host`. Note that the flag cannot be after the hostname.

**Comment by denalex** [comment]
`pxf sync --delete host` gives an impression that we are trying to take a given host out of PXF cluster, just saying ...

**Comment by lisakowen** [comment]
good point, alex.  what about --delete-files instead?

File: cli/go/src/pxf-cli/cmd/cluster.go [hunk 3]
```diff
@@ -111,12 +117,12 @@ func GenerateOutput(command *pxf.Command, clusterData *ClusterData) error {
 		}
 		response += fmt.Sprintf("%s ==> %s\n", host, errorMessage)
 	}
-	gplog.Info("ERROR: "+command.Messages(pxf.Error), numErrors, numHosts)
+	gplog.Info("ERROR: "+cmd.messages[err], numErrors, numHosts)
```
Comment (denalex) [inline cli/go/src/pxf-cli/cmd/cluster.go]: should ERROR be logged at ERROR level rather than INFO ?

File: cli/go/src/pxf-cli/cmd/cluster.go [hunk 4]
```diff
@@ -146,28 +152,28 @@ func adaptContentIDToHostname(cluster *cluster.Cluster, f func(string) string) f
...
 }
 
-func clusterRun(command *pxf.Command, clusterData *ClusterData) error {
+func clusterRun(cmd *command, clusterData *ClusterData) error {
 	defer clusterData.connection.Close()
 
-	err := command.Warn(os.Stdin)
+	err := cmd.Warn(os.Stdin)
 	if err != nil {
 		gplog.Info(fmt.Sprintf("%s", err))
```
Comment (denalex) [inline cli/go/src/pxf-cli/cmd/cluster.go]: should this be Warn level ?

File: cli/go/src/pxf-cli/cmd/cluster.go (hunk 3 above)
Comment (oliverralbertini) [reply cli/go/src/pxf-cli/cmd/cluster.go]: thanks for the feedback, I will address these in an upcoming PR.

File: cli/go/src/pxf-cli/cmd/cluster.go (hunk 4 above)
Comment (oliverralbertini) [reply cli/go/src/pxf-cli/cmd/cluster.go]: see the upcoming PR to adjust these log levels.

**Pull request merged**